### PR TITLE
[Feature] Set ttlSecondsAfterFinished in argo workflow with PipelineConf

### PR DESCRIPTION
**Pull request by eterna2**
### Background
Workflow resources are not removed after kf runs. Some of my runs involves hundreds of pods, which can result in a slow-down in the UI if there are not GC.

I know there is #844. So I am wondering what are the current workarounds? Currently, I am just running a cron job to remove resources > X days from creations.

So I am wondering if it is better to allow the sdk to set the [`ttlSecondsAfterFinished`](https://github.com/argoproj/argo/blob/master/api/openapi-spec/swagger.json#L1245) for the pipeline?

### PR
This PR adds a new method `set_ttl_seconds_after_finished` to `PipelineConf` whereby the compiler will insert a new field `ttlSecondsAfterFinished` to the workflow spec if `ttl_seconds_after_finished >= 0`.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1594)

### CODE REVIEW

File: sdk/python/tests/compiler/compiler_tests.py [hunk 1]
```diff
@@ -334,6 +334,10 @@ def test_py_recursive_while(self):
     """Test pipeline recursive."""
     self._test_py_compile_yaml('recursive_while')
 
+  def test_py_compile_ttl_after_finished(self):
```
Comment (Ark-kun) [inline sdk/python/tests/compiler/compiler_tests.py]: Can you make the test more focused on the actual introduced behavior?
See `test_set_display_name` for example.

Comment (eterna2) [reply sdk/python/tests/compiler/compiler_tests.py]: - Removed the unnecessary compile test.
- Added an unit test instead.

**Comment by Ark-kun** [comment]
/hold
@gaoning777 WDYT?

**Pull request merged**